ci/prow/make_config.go

```diff
@@ -795,9 +923,65 @@ func strExists(arr []string, str string) bool {
 	return false
 }
 
+type specialBranchLogic struct {
+	branches []string
+	// create new job data based on branches
+	opsNew  func(*baseProwJobTemplateData)
+	restore func(*baseProwJobTemplateData)
+}
+
+// getBase casts data into baseProwJobTemplateData and returns it
+func getBase(data interface{}) *baseProwJobTemplateData {
+	var base *baseProwJobTemplateData
+	switch v := data.(type) {
+	case *presubmitJobTemplateData:
+		base = &data.(*presubmitJobTemplateData).Base
+	case *postsubmitJobTemplateData:
+		base = &data.(*postsubmitJobTemplateData).Base
+	default:
+		log.Fatalf("Unrecognized job template type: '%v'", v)
+	}
+	return base
+}
+
+// recursiveSBL recursively going through specialBranchLogic, and generate job at last
+func recursiveSBL(repoName string, data interface{}, generateOneJob func(data interface{}), sbs []specialBranchLogic, i, posCount int) {
```

Comment:
Can you add some comments about what is `i` and what is `posCount`?

Reply:
added, actually realized posCount is not used anymore so removed it